x/ethbridge/types/msgs.go

```diff
@@ -86,26 +124,35 @@ func (msg MsgLock) GetSigners() []sdk.AccAddress {
 
 // MsgBurn defines a message for burning coins and triggering a related event
 type MsgBurn struct {
-	CosmosSender     sdk.AccAddress  `json:"cosmos_sender" yaml:"cosmos_sender"`
-	Amount           sdk.Int         `json:"amount" yaml:"amount"`
-	Symbol           string          `json:"symbol" yaml:"symbol"`
-	EthereumChainID  int             `json:"ethereum_chain_id" yaml:"ethereum_chain_id"`
-	EthereumReceiver EthereumAddress `json:"ethereum_receiver" yaml:"ethereum_receiver"`
+	CosmosSender         sdk.AccAddress  `json:"cosmos_sender" yaml:"cosmos_sender"`
+	CosmosSenderSequence uint64          `json:"cosmos_sender_sequence" yaml:"cosmos_sender_sequence"`
+	Symbol               string          `json:"symbol" yaml:"symbol"`
+	Amount               sdk.Int         `json:"amount" yaml:"amount"`
+	EthereumChainID      int             `json:"ethereum_chain_id" yaml:"ethereum_chain_id"`
+	CethAmount           sdk.Int         `json:"ceth_amount" yaml:"ceth_amount"`
+	MessageType          MessageType     `json:"message_type" yaml:"message_type"`
+	EthereumReceiver     EthereumAddress `json:"ethereum_receiver" yaml:"ethereum_receiver"`
 }
 
 // NewMsgBurn is a constructor function for MsgBurn
 func NewMsgBurn(
 	ethereumChainID int, cosmosSender sdk.AccAddress,
-	ethereumReceiver EthereumAddress, amount sdk.Int, symbol string) MsgBurn {
+	ethereumReceiver EthereumAddress, amount sdk.Int, symbol string, cethAmount sdk.Int, messageType MessageType) MsgBurn {
 	return MsgBurn{
-		EthereumChainID:  ethereumChainID,
-		CosmosSender:     cosmosSender,
-		EthereumReceiver: ethereumReceiver,
-		Amount:           amount,
-		Symbol:           symbol,
+		EthereumChainID:      ethereumChainID,
+		CosmosSender:         cosmosSender,
+		CosmosSenderSequence: 0,
```

Comment:
This cannot be set to 0. We should pass this in as the correct cosmos account sequence.